File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderProductRemovedEvent.java
package ru.quipy.shop.order.events;

import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.order.entities.Product;
import ru.quipy.shop.order.OrderAggregate;

import java.util.UUID;


@DomainEvent(name = "PRODUCT_REMOVED_EVENT")
public class OrderProductRemovedEvent extends Event<OrderAggregate> {

    @Getter
    private final UUID orderId;

    @Getter
    private final Product product;

    public OrderProductRemovedEvent(UUID orderId, Product product) {
        super();
        this.orderId = orderId;
        this.product = product;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderProductRemovedEvent.kt
package ru.quipy.shop.order.events

import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.order.OrderAggregate
import ru.quipy.shop.product.Product

import java.util.UUID


@DomainEvent(name = "PRODUCT_REMOVED_EVENT")
class OrderProductRemovedEvent(
    @Getter
    private val orderId: UUID,
    @Getter val product: Product
) : Event<OrderAggregate>(
    name = "PRODUCT_REMOVED_EVENT"
)